pkg/doc/did/doc.go

```diff
@@ -386,16 +386,17 @@ type Proof struct {
 
 // ParseDocument creates an instance of DIDDocument by reading a JSON document from bytes
 func ParseDocument(data []byte) (*Doc, error) {
-	// validate did document
-	if err := validate(data); err != nil {
-		return nil, err
-	}
-
 	raw := &rawDoc{}
 
 	err := json.Unmarshal(data, &raw)
 	if err != nil {
 		return nil, fmt.Errorf("JSON marshalling of did doc bytes bytes failed: %w", err)
+	} else if raw == nil {
+		return nil, errors.New("document payload is not provided")
+	}
+	// validate did document
+	if err := validate(data, raw.schemaLoader()); err != nil {
```

Comment:
replace the above line with:
```
err = validate(data, raw.schemaLoader())
if err != nil {
```
to remove the govet error

Reply:
Done